airbyte-integrations/connectors/source-google-adwords-singer/source_google_adwords_singer/source.py

```diff
@@ -0,0 +1,59 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from airbyte_protocol import AirbyteConnectionStatus, Status
+from base_singer import SingerSource
+
+
+class SourceGoogleAdwordsSinger(SingerSource):
+    def __init__(self):
+        pass
+
+    def check(self, logger, config_container) -> AirbyteConnectionStatus:
+        try:
```

Comment:
curious if people love or hate this approach to check connection in a singer source. i'm liking it because it spares me having to write custom auth code to test credentials. instead we use exactly what singer will use and don't need to worry about it diverging.

using read and not discover because discover is expensive against the API quota.

Reply:
I think it's fine. I don't like that it's interacting with the integration more than it should but it is easier to implement. Bit of wash.

Reply:
I like that it gives certainty that the impl works as long as we are confident it will error if creds are incorrect